src/MICore/LaunchOptions.cs

```diff
@@ -1055,10 +1100,51 @@ public static LaunchOptions GetInstance(HostConfigurationStore configStore, stri
             if (launchOptions._setupCommands == null)
                 launchOptions._setupCommands = new List<LaunchCommand>(capacity: 0).AsReadOnly();
 
+            // load supplemental options from the solution root
+            string slnRoot = HostNatvisProject.FindSolutionRoot();
+            if (!string.IsNullOrEmpty(slnRoot))
+            {
+                string optFile = Path.Combine(slnRoot, "Microsoft.MIEngine.Options.xml");
+                if (File.Exists(optFile))
+                {
+                    var reader = File.OpenText(optFile);
+                    string suppOptions = reader.ReadToEnd();
+                    if (!string.IsNullOrEmpty(suppOptions))
+                    {
+                        try
+                        {
+                            logger?.WriteTextBlock("SupplementalLaunchOptions", suppOptions);
+                            XmlReader xmlRrd = OpenXml(suppOptions);
+                            XmlSerializer serializer = GetXmlSerializer(typeof(Xml.LaunchOptions.SupplementalLaunchOptions));
+                            var xmlSuppOptions = (Xml.LaunchOptions.SupplementalLaunchOptions)Deserialize(serializer, xmlRrd);
+                            launchOptions.Merge(xmlSuppOptions);
+                        }
+                        catch (Exception e)
+                        {
+                            throw new Exception(string.Format(CultureInfo.CurrentCulture, MICoreResources.Error_ProcessingFile, "Microsoft.MIEngine.Options.xml", e.Message), e);
+                        }
+                    }
+                }
+            }
+
             launchOptions.SetInitializationComplete();
             return launchOptions;
         }
 
+        private void Merge(SupplementalLaunchOptions suppOptions)
+        {
+            // merge the source mapping lists
+            List<PathMapEntry> map = new List<PathMapEntry>(SourceMap);
```

Comment:
>SourceMap [](start = 60, length = 9)

I feel like the new values should go at the front of the list so they take precedence over what was supplied by the project system.